transform/snowflake-dbt/models/mattermost/hourly/user_retention.sql

```diff
@@ -0,0 +1,52 @@
+{{config({
+    "materialized": "incremental",
+    "schema": "mattermost",
+    "tags":"union",
+    "snowflake_warehouse": "transform_m",
+    "unique_key":"id"
+  })
+}}
+
+WITH first_active AS (
+    SELECT
+        uet.user_actual_id
+      , sf.server_id
+      , {{ dbt_utils.surrogate_key(['uet.user_actual_id', 'sf.server_id'])}} AS id
+      , sf.installation_id
+      , sf.first_server_edition
+      , MIN(uet.timestamp) AS first_active_timestamp
+    FROM {{ ref('user_events_telemetry') }} uet
+         JOIN {{ ref('server_fact') }}  sf
+              ON COALESCE(uet.user_id, IFF(LENGTH(uet.context_server) < 26, NULL, uet.context_server),
+                          IFF(LENGTH(uet.context_traits_userid) < 26, NULL, uet.context_traits_userid),
+                          IFF(LENGTH(uet.context_server) < 26, NULL, uet.context_server)) = sf.server_id
+            AND uet.timestamp::date >= sf.first_active_date::date
+            AND sf.first_active_date::date >= '2020-02-01'
```

Comment:
should we move this out as a var/const before, to be referenced?

Reply:
I don't know if that is possible in dbt/sql. we can do it using Stored Procedures, but that would make things really complex